src/components/cursor-controller.js

```diff
@@ -151,12 +151,18 @@ AFRAME.registerComponent("cursor-controller", {
       this.direction.copy(raycaster.ray.direction);
     } else if ((this.inVR || this.isMobile) && !this.hasPointingDevice) {
       //gaze cursor mode
-      camera.getWorldPosition(this.origin);
-      camera.getWorldDirection(this.direction);
+      camera.updateMatrixWorld(true);
+      this.origin.setFromMatrixPosition(camera.matrixWorld);
+      this.controllerQuaternion.setFromRotationMatrix(camera.matrixWorld);
+      this.direction
```

Comment:
Probably fine to do in its own PR since irs maybe too divergent from this perf fix, but we can maybe try and get rid of `this.direction` since it seems like duplicate information, but I havent dug deep enough to know if thats true.